### PR TITLE
Cleanup Hook functions

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Turn HC_ constants into enum
- Cleanup SetWindowsHookEx and made unicode
- Cleanup CallNextHook
- Cleanup UnhookWindowsHookEx

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2035)

### CODE REVIEW

File: src/Common/src/Interop/User32/Interop.SetWindowsHookExW.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [DllImport(Libraries.User32, ExactSpelling = true)]
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.SetWindowsHookExW.cs]: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowshookexw#return-value
```suggestion
        [DllImport(Libraries.User32, ExactSpelling = true, SetLastError = true)]
```

File: src/Common/src/Interop/User32/Interop.UnhookWindowsHookEx.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [DllImport(ExternDll.User32, ExactSpelling = true)]
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.UnhookWindowsHookEx.cs]: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-unhookwindowshookex#return-value

Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.UnhookWindowsHookEx.cs]: Fixed

File: src/Common/src/Interop/User32/Interop.SetWindowsHookExW.cs (hunk 1 above)
Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.SetWindowsHookExW.cs]: Fixed

Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.SetWindowsHookExW.cs]: Fixed

File: global.json [hunk 3]
```diff
@@ -20,6 +20,5 @@
     "Microsoft.NET.Sdk.IL": "5.0.0-alpha1.19501.1"
   },
   "native-tools": {
-    "dotnet-api-docs_netcoreapp3.0": "0.0.0.1"
```
Comment (zsd4yr) [inline global.json]: did you mean to make this change?

Comment (hughbe) [reply global.json]: Nope - removed thanks

**Comment by zsd4yr** [comment]
@msftbot please merge if @RussKie approves

**Pull request merged**